script.js

```diff
@@ -55,23 +39,34 @@ class World {
     gameField.appendChild(object.element);
     this.playFieldObjects.push(object);
   }
+
+  anyCollisionsWith(entity) {
+    const offset = {
+      x: 9,
+      y: 9
+    }
+
+    return this.playFieldObjects.some(playFieldObject => {
+      return this.willCollide(playFieldObject, entity, offset);
+    });
+  }
+
+  willCollide(rect1, rect2, offset) {
+    return rect1.x < rect2.x + rect2.width &&
+          rect1.x + rect1.width > rect2.x &&
+          rect1.y < rect2.y + rect2.height &&
+          rect1.y + rect1.height > rect2.y
+  }
 }
 
 const gameField = document.querySelector('#game');
 
 const player = {
   x: 100,
   y: 400,
-  width: 200,
+  width: 145,
```

Comment:
Curious about this change? The leg is getting clipped when walking. Was it for the collision detection?

Reply:
It was for collision. it can be adjusted a bit too. I'll set the overflow of the image to fix the clipping

Reply:
I have solved this by just setting the width and height of the div in the CSS but using x and y internally for the hitbox calculation